README.md
# Flex Design Colors
**An intentional, transparent, and accessible color system for the modern web.**

---

[![🚀 Build & Publish](https://github.com/kevintyj/color/actions/workflows/publish.yml/badge.svg?branch=main)](https://github.com/kevintyj/color/actions/workflows/publish.yml)
[![🧪 CI Tests](https://github.com/kevintyj/color/actions/workflows/ci.yml/badge.svg?branch=main)](https://github.com/kevintyj/color/actions/workflows/ci.yml)
[![Latest Release](https://img.shields.io/github/v/release/kevintyj/color)](https://github.com/kevintyj/color/releases)
[![License](https://img.shields.io/github/license/kevintyj/color)](https://github.com/kevintyj/color/blob/main/LICENSE)

Accessibility tested color scale library for the modern web and mobile, packaged, and ready to use with any styling 
library. Check out [Ambient Colors](https://github.com/kevintyj/ambient) for more information!

## Getting started
### Install the colors as a dependency 
```bash
pnpm add @kevintyj/color
```
### Use the colors as objects inside any js(x)/ts(x) file
Use any of the colors as objects by importing the color variables
```js
import { BaseBackground, blue } from '@kevintyj/color';

const blueScale = blue;
/*
blueScale = {
    0: '#e9eefe',
    1: '#dddffc',
    2: '#cac9f9',
    3: '#afacf5',
    4: '#7b7ced',
    5: '#5f68e9',
    6: '#264fe3',
    7: '#1a39ac',
    8: '#102478',
    9: '#061148',
};
*/

const backgroundLight = BaseBackground.light;
/*
backgroundLight = '#FFFFFF';
*/
```

Import the types in a typescript porject for typesafety when using the entire scale

```ts
import type { lightColorsType } from '@kevintyj/color';

const colorProp: lightColorsType = 'blue';
/*
typeof colorProp = 'neutral' | 'ocean' | 'blue' | 'indigo'  ...
*/
```

### Use with other styling libraries 
**Use with [tailwind](https://tailwindcss.com)**

`tailwind.config.js`

```js
import { BaseBackground, blue, blueDark } from '@kevintyj/color';

/** @type {import('tailwindcss').Config} */
export default {
  // ...
  theme: {
    extend: {
      colors: {
        BaseBackground,
        blue,
        blueDark
      },
    },
  },
  // ...
};
```

`component.html`

```html
<div class="bg-blue-5">This div has a wonderfully blue color!</div>
```

You can also import all the colors at once.

> This method of importing colors imports both the light and dark colors

`tailwind.config.js`

```js
import * as colors from '@kevintyj/color';

/** @type {import('tailwindcss').Config} */
export default {
  // ...
  theme: {
    extend: {
      colors: {
        ...colors
      },
    },
  },
  // ...
};
```

**Use with [unocss](https://tailwindcss.com)**

`uno.config.ts`

```ts
import {
  defineConfig,
} from 'unocss'
import { BaseBackground, blue, blueDark } from '@kevintyj/color';

export default defineConfig({
  // ...
  theme: {
    colors: {
      BaseBackground,
      blue,
      blueDark
    },
  },
  // ...
});
```

`component.html`

```html
<div class="bg-blue-5">This div has a wonderfully blue color!</div>
```

You can also import all the colors at once by calling `import * as colors from '@kevintyj/color'` and calling 
`...colors` in the theme config.

## License

Licensed under the BSD-3 License, Copyright © 2023-present [Kevin Taeyoon Jin](https://github.com/kevintyj).

See [LICENSE](./LICENSE) for more information


CHANGELOG.md
# @kevintyj/color

## 0.2.1

### Patch Changes

- [`8001e8c`](https://github.com/kevintyj/color/commit/8001e8c3cca5a5e2c434540b2dc9aa3ec695738a) Thanks [@kevintyj](https://github.com/kevintyj)! - Add light/dark js exports

## 0.2.0

### Minor Changes

- [`2a593b8`](https://github.com/kevintyj/color/commit/2a593b8882b92edcca888bc4aecd9556a6779dfc) Thanks [@kevintyj](https://github.com/kevintyj)! - Generate CSS exports

- [`5cfdeb6`](https://github.com/kevintyj/color/commit/5cfdeb6bc460c591571a4f72d9bdfd2eb0931d98) Thanks [@kevintyj](https://github.com/kevintyj)! - Generate SCSS exports

### Patch Changes

- [`5cfdeb6`](https://github.com/kevintyj/color/commit/5cfdeb6bc460c591571a4f72d9bdfd2eb0931d98) Thanks [@kevintyj](https://github.com/kevintyj)! - Add BaseColors (white/black) to colorset

## 0.1.1

### Patch Changes

- [#5](https://github.com/kevintyj/color/pull/5) [`97b5a97`](https://github.com/kevintyj/color/commit/97b5a970c8fc75ff2d6ca1d3a5c1a390caa673fc) Thanks [@kevintyj](https://github.com/kevintyj)! - Add Flex Color Uniform as string literal types

## 0.1.0

### Minor Changes

- [#1](https://github.com/kevintyj/color/pull/1) [`7019b15`](https://github.com/kevintyj/color/commit/7019b158d2fba5e423a7dd211ea92f4b587518ba) Thanks [@kevintyj](https://github.com/kevintyj)! - Add Flex Colors Uniform as TS objects

  Colors generated using [Ambient Colors](https://github.com/kevintyj/ambient)


src/light.ts
export const neutral: ColorScale<HexColor> = {
	0: '#f8f8f6',
	1: '#eaeae8',
	2: '#d5d5d4',
	3: '#babab9',
	4: '#8d8d8c',
	5: '#7a7a7a',
	6: '#626262',
	7: '#484847',
	8: '#2f2f2d',
	9: '#181816',
};

export const ocean: ColorScale<HexColor> = {
	0: '#e6f5fa',
	1: '#dbedfa',
	2: '#c9e0fb',
	3: '#b0d0fb',
	4: '#80b4fb',
	5: '#66a9fa',
	6: '#359afa',
	7: '#256eba',
	8: '#16457d',
	9: '#081f46',
};

export const blue: ColorScale<HexColor> = {
	0: '#e9eefe',
	1: '#dddffc',
	2: '#cac9f9',
	3: '#afacf5',
	4: '#7b7ced',
	5: '#5f68e9',
	6: '#264fe3',
	7: '#1a39ac',
	8: '#102478',
	9: '#061148',
};

export const indigo: ColorScale<HexColor> = {
	0: '#ebedfb',
	1: '#e0dbf8',
	2: '#cfc1f4',
	3: '#b69eed',
	4: '#8763e1',
	5: '#6f49db',
	6: '#4624d3',
	7: '#30199e',
	8: '#1a0e6b',
	9: '#0b003d',
};

export const violet: ColorScale<HexColor> = {
	0: '#f0e9f8',
	1: '#e7d8f6',
	2: '#d9bff3',
	3: '#c59dee',
	4: '#9d62e5',
	5: '#8a47e1',
	6: '#6c1cdb',
	7: '#5015a3',
	8: '#350e6e',
	9: '#1c063d',
};

export const crimson: ColorScale<HexColor> = {
	0: '#faeaf2',
	1: '#fadce8',
	2: '#f9c7d8',
	3: '#f6abc4',
	4: '#ed77a1',
	5: '#e85f93',
	6: '#e03780',
	7: '#a8255b',
	8: '#731438',
	9: '#420518',
};

export const wine: ColorScale<HexColor> = {
	0: '#fde9ed',
	1: '#f7d7db',
	2: '#edbdc1',
	3: '#de9ba0',
	4: '#c05f68',
	5: '#b24451',
	6: '#9d1635',
	7: '#770d27',
	8: '#53051a',
	9: '#32000a',
};

export const red: ColorScale<HexColor> = {
	0: '#feeeec',
	1: '#fddeda',
	2: '#fac6c0',
	3: '#f3a69e',
	4: '#e26c65',
	5: '#d9514e',
	6: '#ca2331',
	7: '#961524',
	8: '#650717',
	9: '#380004',
};

export const sun: ColorScale<HexColor> = {
	0: '#fef2ee',
	1: '#ffe4dc',
	2: '#ffcec3',
	3: '#ffb1a1',
	4: '#f97c68',
	5: '#f36451',
	6: '#ea3e33',
	7: '#af2925',
	8: '#781617',
	9: '#460405',
};

export const orange: ColorScale<HexColor> = {
	0: '#fdf6ed',
	1: '#ffebda',
	2: '#ffdbbf',
	3: '#ffc69b',
	4: '#ffa05d',
	5: '#ff8f41',
	6: '#fb7912',
	7: '#bf5614',
	8: '#873512',
	9: '#52160b',
};

export const amber: ColorScale<HexColor> = {
	0: '#fdfaf2',
	1: '#fff3e1',
	2: '#ffe9c6',
	3: '#ffdba4',
	4: '#ffc467',
	5: '#ffba4b',
	6: '#fdad18',
	7: '#c17c1a',
	8: '#894d17',
	9: '#542210',
};

export const yellow: ColorScale<HexColor> = {
	0: '#fdfcf2',
	1: '#fff7e1',
	2: '#fff0c8',
	3: '#ffe6a7',
	4: '#ffd56b',
	5: '#fdce4e',
	6: '#f8c51a',
	7: '#c08d1a',
	8: '#8b5916',
	9: '#58280f',
};

export const sapling: ColorScale<HexColor> = {
	0: '#fafeec',
	1: '#f3f9dd',
	2: '#e9f2c7',
	3: '#dae9aa',
	4: '#bfd977',
	5: '#b3d261',
	6: '#a2c940',
	7: '#78962b',
	8: '#506618',
	9: '#2b3a06',
};

export const green: ColorScale<HexColor> = {
	0: '#eefdf3',
	1: '#def6e4',
	2: '#c6ebce',
	3: '#a6dcb1',
	4: '#6bc280',
	5: '#4db66b',
	6: '#0da750',
	7: '#067c39',
	8: '#015423',
	9: '#002f0f',
};

export const aurora: ColorScale<HexColor> = {
	0: '#ebfdf9',
	1: '#dcf7f0',
	2: '#c5ede2',
	3: '#a6e0d0',
	4: '#6bc9b0',
	5: '#4ebfa3',
	6: '#02b192',
	7: '#01846c',
	8: '#005949',
	9: '#003228',
};

export const cyan: ColorScale<HexColor> = {
	0: '#effdfd',
	1: '#e2f8fa',
	2: '#cef1f5',
	3: '#b2e7ef',
	4: '#7bd6e4',
	5: '#5ecedf',
	6: '#20c4d9',
	7: '#148fa2',
	8: '#085e6f',
	9: '#01313f',
};
